src/components/Modal/ConfirmationModal.js

```diff
@@ -1,46 +1,68 @@
 import React from "react";
+import { useDispatch, useSelector } from "react-redux";
 import styled from "styled-components";
-import PropTypes from "prop-types";
 
 import Portal from "../Portal/Portal";
 import Button from "../shared/Button";
 import catchAsync from "../../utils/catchAsync";
+import {
+  fetchGroupInfo,
+  fetchUserInfo,
+  setModalInfo,
+} from "../../features/todoSlice";
 import * as api from "../../api";
 
-const ConfirmationModal = ({ setModalOn }) => {
-  const propsCategory = "";
-  const groupId = "";
-  const todoId = "";
+const ConfirmationModal = () => {
+  const dispatch = useDispatch();
+  const modalInfo = useSelector((state) => state.modalInfo);
 
-  const handleSubmit = catchAsync(async () => {
-    switch (propsCategory) {
-      case "Delete TODO":
-        await api.deleteTodo(todoId);
+  const closeModal = () => {
+    dispatch(
+      setModalInfo({
+        propsCategory: modalInfo.propsCategory.includes("Group")
+          ? "Group"
+          : "TODO",
+      })
+    );
+  };
+
+  const handleSubmit = catchAsync(async (event) => {
+    event.preventDefault();
+
+    switch (modalInfo.propsCategory) {
+      case "DeleteTODO":
+        await api.deleteTodo(modalInfo.todoId);
         break;
-      case "Delete Group TODO":
-        await api.deleteGroupTodo(groupId, todoId);
+      case "DeleteGroupTODO":
+        await api.deleteGroupTodo(modalInfo.groupId, modalInfo.todoId);
         break;
-      case "Delete Group":
-        await api.deleteGroup(groupId);
+      case "DeleteGroup":
+        await api.deleteGroup(modalInfo.groupId);
         break;
       default:
     }
 
-    setModalOn(false);
+    if (modalInfo.groupId && modalInfo.propsCategory.includes("TODO")) {
+      dispatch(fetchGroupInfo(modalInfo));
+    } else {
+      dispatch(fetchUserInfo());
+    }
+
+    closeModal();
   });
 
   return (
     <Portal>
       <Background>
         <Content>
-          <MessageParagraph>props message</MessageParagraph>
+          <MessageParagraph>{modalInfo.confirmMessage}</MessageParagraph>
           <div>
             <ResponseButton onClick={handleSubmit}>YES</ResponseButton>
-            <ResponseButton onClick={() => setModalOn(false)}>
-              NO
-            </ResponseButton>
+            <ResponseButton onClick={() => closeModal()}>NO</ResponseButton>
```

Comment:
다은님! line 61에서 line 60처럼 함수를 호출하는 방식을 통일해주셔도 좋을 것 같아요!
```suggestion
            <ResponseButton onClick={closeModal}>NO</ResponseButton>
```

Reply:
네! 바로 수정하도록 하겠습니다!